vyper/semantics/validation/module.py

```diff
@@ -202,6 +205,9 @@ def visit_AnnAssign(self, node):
             annotation, data_loc, is_constant, is_public, is_immutable
         )
         node._metadata["type"] = type_definition
+        node.is_constant = is_constant
```

Comment:
would it be better to calculate these properties during node construction? they are more properties of the AST (with the new definition of VariableDef) than things that need to be analyzed

Reply:
yes, that's a good point. I will make the changes. thanks!